diff --git a/node_modules/@wordpress/api-fetch/build-module/utils/response.js b/node_modules/@wordpress/api-fetch/build-module/utils/response.js
index e7affec..56810e8 100644
--- a/node_modules/@wordpress/api-fetch/build-module/utils/response.js
+++ b/node_modules/@wordpress/api-fetch/build-module/utils/response.js
@@ -19,7 +19,14 @@ var parseResponse = function parseResponse(response) {
       return null;
     }
 
-    return response.json ? response.json() : Promise.reject(response);
+    var rclone = response.clone();
+    return response.json ? response.json().catch(function (e) {
+      rclone.text().then((text) => console.error('invalid_json_raw', response.url, text));
+      throw e;
+    }) : Promise.reject(response).catch(function (e) {
+      rclone.text().then((text) => console.error('invalid_response_raw', response.url, text));
+      throw e;
+    });
   }
 
   return response;
diff --git a/node_modules/@wordpress/api-fetch/build/utils/response.js b/node_modules/@wordpress/api-fetch/build/utils/response.js
index cb7154b..8025ba1 100644
--- a/node_modules/@wordpress/api-fetch/build/utils/response.js
+++ b/node_modules/@wordpress/api-fetch/build/utils/response.js
@@ -28,7 +28,14 @@ var parseResponse = function parseResponse(response) {
       return null;
     }
 
-    return response.json ? response.json() : Promise.reject(response);
+    var rclone = response.clone();
+    return response.json ? response.json().catch(function (e) {
+      rclone.text().then((text) => console.error('invalid_json_raw', response.url, text));
+      throw e;
+    }) : Promise.reject(response).catch(function (e) {
+      rclone.text().then((text) => console.error('invalid_response_raw', response.url, text));
+      throw e;
+    });
   }
 
   return response;
